File: pgm/ABC_Java/ABC_416_A.java
import java.io.*;
// import java.math.*;
// import java.time.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final String S = reader.readLine();
        reader.close();
        final String Yes = "Yes";
        final String No = "No";
        final String circle = "o";
        final String plus = "+";
        final int one = 1;
        final int two = 2;
        final int L = Integer.parseInt (arr[one]);
        final int R = Integer.parseInt (arr[two]);
        final String pattern = circle.concat (plus);
        System.out.println (S.substring (L - one, R).matches (pattern) ? Yes
                                                                       : No);
        System.exit (0);
    }
}

File: pgm/ABC_Kotlin/ABC_416_A.kt
// import java.time.*

// import kotlin.math.*

const val circle: String = "o"
const val Yes: String = "Yes"
const val No: String = "No"
const val space: String = " "
const val one: Int = 1

fun main() {
    val (_, L: Int, R: Int) = readln().split(space).map { it.toInt() }
    val S: String = readln()
    val reg: String = """^$circle+$"""
    if (reg.toRegex().matches(S.substring(L - one, R))) {
        println(Yes)
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}